sklearn/datasets/tests/test_base.py

```diff
@@ -121,6 +131,29 @@ def test_load_files_wo_load_content(
     assert res.get('data') is None
 
 
+def test_load_files_w_allowed_and_ignored_extensions(load_files_root):
+    msg = "Ignored extensions and allowed extensions cannot both be present."
+    " Please choose one or the other."
+
+    with pytest.raises(AssertionError, match=msg):
+        load_files(load_files_root, allowed_extensions=[".txt"],
+                   ignored_extensions=[".txt"])
+
+
+def test_load_files_w_ignore_list(test_category_dir_1, test_category_dir_3,
+                                  load_files_root):
+    res = load_files(load_files_root, ignored_extensions=[".txt"])
+    assert len(res.filenames) == 1
+    assert not all([re.search(r".*\.txt$", f) for f in res.filenames])
```

Comment:
I know the rest of the file uses these other fixtures, but moving forward, I would prefer us to use the [tmp_path](https://docs.pytest.org/en/stable/tmpdir.html) fixture. We can create a temporary directory and have some 2 txt + 1 json files in them, and call `load_files` with different `ignored_extensions` and `allowed_extensions` and assert the expected behavior.